File: src/main/java/com/nurflugel/dependencyvisualizer/ui/NurflugelDialog.java
/*
 * Copyright (c) 2006, Your Corporation. All Rights Reserved.
 */
package com.nurflugel.dependencyvisualizer.ui;

import javax.swing.*;
import java.awt.*;

/**  */
@SuppressWarnings({ "AbstractClassExtendsConcreteClass" })
public abstract class NurflugelDialog extends JDialog {
  protected abstract void addListeners();

  /**  */
  void center() {
    
    String alpha="abcedfg";
    String dibble="adfafd";
    boolean allMatch    = dibble.chars().allMatch(d -> alpha.indexOf(d) >= 0);

    Toolkit   defaultToolkit = Toolkit.getDefaultToolkit();
    Dimension screenSize     = defaultToolkit.getScreenSize();
    int       x              = (int) ((screenSize.getWidth() - getWidth()) / 2);
    int       y              = (int) ((screenSize.getHeight() - getHeight()) / 2);

    setBounds(x, y, getWidth(), getHeight());
  }
 /**  */
  boolean center2() {
    
    Character ch=new Character(((char) 5));
    
    String alpha="abcedfg";
    String text="adfafd";
    boolean allMatch    = alpha.chars().allMatch(d -> text.indexOf(d) >= 0);
return allMatch;
  }

  void setHeightToScreen() {
    Toolkit   defaultToolkit = Toolkit.getDefaultToolkit();
    Dimension screenSize     = defaultToolkit.getScreenSize();
    int       width          = (int) getSize().getWidth();
    int       height         = (int) screenSize.getHeight();

    setSize(width, height);
  }

  void setHeightToHalfScreen() {
    Toolkit   defaultToolkit = Toolkit.getDefaultToolkit();
    Dimension screenSize     = defaultToolkit.getScreenSize();
    int       width          = (int) getSize().getWidth();
    int       height         = (int) screenSize.getHeight() / 2;

    setSize(width, height);
  }
}

File: src/main/kotlin/com/nurflugel/dependencyvisualizer/ui/NurflugelDialog.kt
package com.nurflugel.dependencyvisualizer.ui

import java.awt.Toolkit
import javax.swing.JDialog

/**   */
abstract class NurflugelDialog : JDialog() {
    protected abstract fun addListeners()

    /**   */
    fun center() {
        val defaultToolkit = Toolkit.getDefaultToolkit()
        val screenSize = defaultToolkit.screenSize
        val x = ((screenSize.getWidth() - width) / 2).toInt()
        val y = ((screenSize.getHeight() - height) / 2).toInt()

        setBounds(x, y, width, height)
    }

    fun setHeightToScreen() {
        val defaultToolkit = Toolkit.getDefaultToolkit()
        val screenSize = defaultToolkit.screenSize
        val width = size.getWidth().toInt()
        val height = screenSize.getHeight().toInt()

        setSize(width, height)
    }

    fun setHeightToHalfScreen() {
        val defaultToolkit = Toolkit.getDefaultToolkit()
        val screenSize = defaultToolkit.screenSize
        val width = size.getWidth().toInt()
        val height = screenSize.getHeight() * .7

        setSize(width, height.toInt())
    }
}
